### ISSUE TITLE
HTML attributes shouldn't accept element and component values

### ISSUE THREAD

**Issue by boris-petrov** [typed-ember/glint#214]
Not sure about other attributes but at least the `title` is not checked:

```hbs
<iframe src="" title={{123}}>
</iframe>
```

This should give an error for `title` but it does not.

**Comment by dfreeman** [comment]
We allow setting any primitive value as an HTML attribute, since they typically have reasonable coercions (e.g. `<input value={{this.aNumber}} ...>`) or type-specific behavior (e.g. a boolean value will set or unset the attribute entirely).

If we had richer types available for the attributes that specific elements accept (e.g. `title` is always a string, `disabled` always a boolean, etc), we might consider tighter bounds, but since that type info isn't readily available and would be a huge maintenance burden to maintain by hand, we opted for the current system.

**Comment by dfreeman** [comment]
> I was setting an HTMLElement (the result from a helper) and there was [Object object] in the title

That's actually something I've been meaning to fix! 🙂

Currently we have a single mechanism that encompasses emitting values as either top-level content and or in HTML attributes, but `HTMLElements` are a great example of something that should only be allowed at the top level. Block-less components are another.
